web/src/beta/features/Editor/Visualizer/hooks.ts

```diff
@@ -240,6 +245,19 @@ export default ({
     return !!sceneProperty?.experimental?.experimental_sandbox;
   }, [sceneProperty]);
 
+  const [layersData, setLayersData] = useState(layers);
+
+  useEffect(() => {
+    const handleDisplayLayer = () => {
+      const filteredLayers = layers?.filter(layer => currentPage?.layersIds?.includes(layer.id));
+      const layerPages = (currentPage?.layersIds?.length ?? 0) > 0 ? filteredLayers : [];
+      const results = showStoryPanel ? layerPages : layers;
+      setLayersData(results);
```

Comment:
I don't think we want to filter layers ALWAYS when we might not need to.
Can you check 
```
if (!!showStoryPanel) { 
   setLayersData(layers);
   return;
   };
``` 
at the start of the useEffect so we can avoid the processing 🙏🏻

Reply:
@KaWaite this filter help to obtain the layers that are available in selected page cause in a page we only have array of layersId